src/functions-and-arrays.js

```diff
@@ -1,41 +1,118 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1
+  }
+  else return num2
+}
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
-
+function findLongestWord(words) {
+  let longestWord = words[0];
+  let result;
+  if (words.length === 0) {
+    return null
+  }
+  for (let i = 0; i< words.length; i++) {
+   if (words[i].length > longestWord.length) {
+          longestWord = words[i];     
+   }
+  }
+  return longestWord;
+}
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbers) {
+  let result = 0;
+  for (let i = 0; i < numbers.length; i++) {
+          result += numbers[i]
+  }
+  return result
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(wordsArr) {
+  let averageArr = 0; 
+  for (let i = 0; i < wordsArr.length; i++) {
+   if ( wordsArr[i] === true)  {
+     averageArr += 1
+   }
+   else if  (typeof wordsArr[i] === "string") {
+     averageArr += wordsArr[i].length 
+   }
+   else if (typeof wordsArr[i] === "number") {
+     averageArr += wordsArr[i]
+   }
+ 
+}
```

Comment:
Here is the syntax to throw an error. However you might also want to check for the case if array element is boolean :)

```
else {
      throw new Error("Unsupported data type sir or ma'am");
    }
```